src/tye_lab_to_nwb/neurotensin_valence/images/neurotensinconfocalimagesdatainterface.py

```diff
@@ -37,55 +37,70 @@ def __init__(
         """
         self.verbose = verbose
         self.oif = OifFile(file_path)
-        self.composite_images = tifffile.memmap(composite_tif_file_path, mode="r")
+        self.composite_images = None
+        if composite_tif_file_path:
+            self.composite_images = tifffile.memmap(composite_tif_file_path, mode="r")
 
         super().__init__(file_path=file_path)
 
     def get_metadata(self) -> dict:
         metadata = super().get_metadata()
 
-        metadata_from_file = dict()
+        metadata_from_oif_file = dict()
         settings = self.oif.filesystem.settings
 
         # Add session_start_time
         session_start_time = settings["Acquisition Parameters Common"]["ImageCaputreDate"]
         metadata["NWBFile"].update(session_start_time=datetime.fromisoformat(session_start_time))
 
         image_file_name = settings.name  # 'H28PVT_40x.oif'
-        subject_id, location = re.search("([a-zA-Z]+\d+)([^_]+)", image_file_name).groups()
-        metadata_from_file["Subject"] = dict(subject_id=subject_id)
+        file_name_regex_result = re.search("([a-zA-Z]+\d+)([^_]+)", image_file_name)
+        if file_name_regex_result is not None:
+            subject_id, location = re.search("([a-zA-Z]+\d+)([^_]+)", image_file_name).groups()
+            metadata_from_oif_file["Subject"] = dict(subject_id=subject_id)
 
         software_name = settings["Version Info"]["SystemName"]  # 'FLUOVIEW FV1000'
         software_version = settings["Version Info"]["FileVersion"]  # '1.2.6.0'
         # num_axis = settings["Axis Parameter Common"]["AxisCount"]  # 4
         # axis_order = self.oif.axes  # 'ZCYX' depth, ch, width, height
 
         depth_settings = settings["Axis 3 Parameters Common"]
-        depth_range = np.arange(
-            depth_settings["StartPosition"],
-            depth_settings["EndPosition"] + depth_settings["Interval"],
-            depth_settings["Interval"],
-        )
-        depth_range /= 1e6  # nm to m
-        metadata_from_file["Images"] = dict(
+        depth_range = []
+        if all([pos in depth_settings for pos in ["StartPosition", "EndPosition", "Interval"]]):
+            start = depth_settings["StartPosition"]
+            stop = depth_settings["EndPosition"]
+            step = depth_settings["Interval"]
+            if start > stop:
+                step = -step
+
+            depth_range = np.arange(start, stop + step, step)
+            if "UnitName" in depth_settings:
+                if depth_settings["UnitName"] == "nm":
+                    depth_range /= 1e9  # nm to m
```

Comment:
@CodyCBakerPhD 
The "problem" with that file was that I assumed that "StartPosition" which indicates the "z" position for the images would be always smaller than "EndPosition". 

There is another way to know the depth for each TIF file, which could be more robust than even this approach. For each TIF file there is a`.pty`file, which also has "Axis 3 Parameters" setting where the depth information could be extracted (double-checking that the field exists).

example from a `.pty` file:
```
[Axis 3 Parameters]
AbsPositionUnitName="nm"
AbsPositionValue=47480.0
```